ibis/backends/pandas/execution/maps.py

```diff
@@ -148,14 +149,14 @@ def safe_keys(mapping):
     result = safe_method(mapping, 'keys')
     if result is None:
         return None
-    return list(result)
+    return np.array(list(result))
```

Comment:
Hmm..why do we need to call list here instead of `np.array(result)`? What's the type of result?

Reply:
`result` is an iterable (type is `dict_keys`)—`list(...)` is pretty much just to unpack the iterable. There is `np.fromiter` but it seems to not do type inference (requires `dtype` parameter)

Reply:
Please add a comment